libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -468,6 +393,89 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 	_variable.annotation().type = type;
 }
 
+void ReferencesResolver::operator()(yul::FunctionDefinition const& _function)
+{
+	m_yulInsideFunction = true;
+	this->operator()(_function.body);
+	m_yulInsideFunction = false;
+}
+
+void ReferencesResolver::operator()(yul::Identifier const& _identifier)
+{
+	bool isSlot = boost::algorithm::ends_with(_identifier.name.str(), "_slot");
+	bool isOffset = boost::algorithm::ends_with(_identifier.name.str(), "_offset");
+
+	auto declarations = m_resolver.nameFromCurrentScope(_identifier.name.str());
+	if (isSlot || isOffset)
+	{
+		// special mode to access storage variables
+		if (!declarations.empty())
+			// the special identifier exists itself, we should not allow that.
```

Comment:
No error here?

Reply:
well, the code was moved, not changed (except for context required adjustments). So this is no change in behavior...

Reply:
Do we have a test for this?
I think this is basically this case:
```
contract C {
    uint256 a;
    function f() public pure {
        uint256 a_slot;
        assembly {
            pop(a_slot)
        }
    }
}
```
And since we return here, this will trigger "Identifier not found" on ``a_slot`` later... but we might actually put up a nicer error message here...

Reply:
your example will indeed cause this part of the code to be run